pkg/collector/scheduler.go

```diff
@@ -190,7 +190,16 @@ func (s *CheckScheduler) getChecks(config integration.Config) ([]check.Check, er
 			log.Debugf("Loading check instance for check '%s' using default loaders", config.Name)
 		}
 
-		for _, loader := range s.loaders {
+		// TODO: Remove this special case to use Core loader by default for SNMP
+		loaderList := s.loaders
+		if config.Name == "snmp" && selectedInstanceLoader == "" {
+			loaderList = slices.Clone(s.loaders)
+			if len(loaderList) == 2 && loaderList[0].Name() == "python" && loaderList[1].Name() == "core" {
+				loaderList[0], loaderList[1] = loaderList[1], loaderList[0]
+			}
```

Comment:
You can make the code a bit simpler
```suggestion
			if len(loaderList) == 2 && loaderList[0].Name() == "python" && loaderList[1].Name() == "core" {
				loaderList = []check.Loader{loaderList[1], loaderList[0]}
			}
```

Reply:
Done in [b6e48e3](https://github.com/DataDog/datadog-agent/pull/36207/commits/b6e48e355d7a0b29516ff07677482af65c035cda)